quickwit/quickwit-cluster/src/cluster.rs

```diff
@@ -246,22 +232,45 @@ impl Cluster {
         Ok(cluster)
     }
 
-    /// Return [WatchStream] for monitoring change of `members`
-    pub fn member_change_watcher(&self) -> WatchStream<Vec<ClusterMember>> {
-        WatchStream::new(self.members.clone())
+    /// Returns a [`WatchStream`] for monitoring change of ready `members`.
+    /// Note that it does not monitor changes of properties of members, this means
+    /// that a [`ClusterMember`] has no guarantee to have its properties up-to-date.
+    /// To get the latest properties of a [`ClusterMember`], use
+    /// [`Self::ready_members_from_current_state()`].
+    pub fn ready_member_change_watcher(&self) -> WatchStream<Vec<ClusterMember>> {
+        WatchStream::new(self.members_rx.clone())
+    }
+
+    /// Returns the last [`ClusterMember`] send by chitchat.
+    /// Note that a [`ClusterMember`] has no guarantee to have its properties up-to-date.
+    /// To get the latest properties of a [`ClusterMember`], use
+    /// [`Self::ready_members_from_current_state()`].
+    pub async fn ready_members(&self) -> Vec<ClusterMember> {
+        self.members_rx.borrow().clone()
     }
 
-    /// Return `members` list
-    pub fn members(&self) -> Vec<ClusterMember> {
-        self.members.borrow().clone()
+    /// Returns ready [`ClusterMember`]s built directly from the current state of the cluster.
+    pub async fn ready_members_from_cluster_state(&self) -> Vec<ClusterMember> {
+        let cluster_state = self.state().await;
+        let mut ready_nodes = cluster_state.ready_nodes.clone();
+        let is_self_node_ready = cluster_state
```

Comment:
ok I admit, this is ugly. Adding a method on `ClusterState` would make it nicer but for now `ClusterState` is just used for the API so I did not want to "enrich" it right now.